### ISSUE TITLE
Use Darwin Notifications for snapshots?

### ISSUE THREAD

**Issue by olarivain** [fastlane/fastlane#6523]
This isn't an actual issue, more of a suggestion.
Reading through https://github.com/fastlane/fastlane/tree/master/snapshot#how-does-it-work, I'm wondering if you ever considered using darwin notification center to trigger the snapshot on the app side? If you did, what problem did you run into that made you go down the current way of grabbing screenshots?

In case you didn't, here's a good example of it in action: http://www.matrixprojects.net/p/triggering-events-while-ui-testing/
I just ran their sample code and had good success in the sim/iOS 10/Xcode 8.

Seems a whole lot simpler than scrounging the derived data folder for filenames.

**Comment by TKBurner** [comment]
@olarivain Very interesting! We'll check this out. I'm going to leave it open as a feature request. Thanks!

**Comment by KrauseFx** [comment]
Interesting, thank for letting us know. I wasn't aware of this approach, however I don't see a reason changing this in _snapshot_, as it works without problems with the current approach 👍 
I appreciate you letting us know about this, in case something breaks in the future, this would be a great solution.

**Comment by olarivain** [comment]
Fwiw, I've now implemented this in our app.
Huge gotcha is that you cannot pass any data at all with this method. I tried registering for all notifications, and then embed the screenshot name in the notification name (e.g "screenshot|my_screenshot").
That's not an option. Darwin notification are a very basic IPC system.

I got around it by passing the screenshot name
In environment variables (see XCUIApplication launch arg/environment). We reset the screenshot name at each test. It works for us, but it's a big limitation of the approach.
